cmd/f3/run.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/filecoin-project/go-f3"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/sim/signing"
+	leveldb "github.com/ipfs/go-ds-leveldb"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/discovery/mdns"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+const DiscoveryTag = "f3-standalone"
+
+var runCmd = cli.Command{
+	Name:  "run",
+	Usage: "starts a f3 node",
+	Flags: []cli.Flag{
+		&cli.Uint64Flag{
+			Name:  "id",
+			Value: 0,
+		},
+	},
+	Action: func(c *cli.Context) error {
+		ctx := c.Context
+		h, err := libp2p.New(libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/0/quic-v1"))
+		if err != nil {
+			return xerrors.Errorf("creating libp2p host: %w", err)
+		}
+
+		ps, err := pubsub.NewGossipSub(ctx, h)
+		if err != nil {
+			return xerrors.Errorf("creating gossipsub: %w", err)
+		}
+
+		err = setupDiscovery(h)
+		if err != nil {
+			return xerrors.Errorf("setting up discovery: %w", err)
+		}
+
+		tmpdir, err := os.MkdirTemp("", "f3-*")
```

Comment:
Where does `tmpdir` gets removed?

Reply:
It doesn't help examination when it is used; in the future, it will be a long-lived directory.